### ISSUE TITLE
Specific sender restriction (authOrig) not synced 

### ISSUE THREAD

**Issue by TPok1707** [timmcmic/DLConversionV2#108]
After checking the results of my first sync batch  I noticed that there may be a bug.
We have distribution lists where sending mails should only permitted for specific users. This is stored in the authOrig attribute on premise. The script seems to read this information correctly. In originalDLConfigurationADXML.xml I can see:

```
    <Obj N="authOrig" RefId="1">
        <TN RefId="1">
          <T>Microsoft.ActiveDirectory.Management.ADPropertyValueCollection</T>
          <T>System.Collections.CollectionBase</T>
          <T>System.Object</T>
        </TN>
        <LST>
          <S>CN=Some User1,OU=Normal,OU=Users,DC=ad,DC=company,DC=com</S>
          <S>CN=Some User2,OU=Normal,OU=Users,DC=ad,DC=company,DC=com</S>
          <S>CN=Some User3,OU=Normal,OU=Users,DC=ad,DC=company,DC=com</S>
         </LST>
      </Obj>
```

After migration to the cloud this setting is lost. Anybody is allowed to send to the group.

I went a little bit through the source code. Could this be the area that leads to the error?
File: DLConversionV2.psm1

```
    Out-LogFile -string "Invoke get-NormalizedDN to normalize the accept members DN to Office 365 identifier."

    Out-LogFile -string "ACCEPT USERS"

    if ($originalDLConfiguration.($onPremADAttributes.onPremRejectMessagesFromDLMembers.value) -ne $NULL)
    {
        foreach ($DN in $originalDLConfiguration.($onPremADAttributes.onPremAcceptMessagesFromSenders.value))
        {
            if ($forLoopCounter -eq $forLoopTrigger)
            {
                start-sleepProgress -sleepString "Throttling for 5 seconds..." -sleepSeconds 5
```

Should it be instead like this?
`    if ($originalDLConfiguration.($onPremADAttributes.onPremAcceptMessagesFromSenders.value) -ne $NULL)
`

**Comment by timmcmic** [comment]
Can you send me the full log.  Dlconversionv2 @ service  dot Microsoft dot com


==============================
Timothy J. McMichael
Senior Support Escalation Engineer
***@***.******@***.***>
(980)-776-7465

Hours:  Sunday – Wednesday 08:00 – 16:00 eastern time zone.

Manager:  Tom Roughley ***@***.******@***.***>)

Premier Support - (800)-936-3100
Broad Commercial Support - (800)-936-4900

==============================

________________________________
From: TPok1707 ***@***.***>
Sent: Thursday, February 2, 2023 4:40 PM
To: timmcmic/DLConversionV2 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [timmcmic/DLConversionV2] Specific sender restriction (authOrig) not synced (Issue #108)


After checking the results of my first sync batch I noticed that there may be a bug.
We have distribution lists where sending mails should only permitted for specific users. This is stored in the authOrig attribute on premise. The script seems to read this information correctly. In originalDLConfigurationADXML.xml I can see:

<Obj N="authOrig" RefId="1"> <TN RefId="1"> <T>Microsoft.ActiveDirectory.Management.ADPropertyValueCollection</T> <T>System.Collections.CollectionBase</T> <T>System.Object</T> </TN> <LST> <S>CN=Some User1,OU=Normal,OU=Users,DC=ad,DC=company,DC=com</S> <S>CN=Some User2,OU=Normal,OU=Users,DC=ad,DC=company,DC=com</S> <S>CN=Some User3,OU=Normal,OU=Users,DC=ad,DC=company,DC=com</S> </LST> </Obj>

After migration to the cloud this setting is lost. Anybody is allowed to send to the group.
I could not find any log entry about this indicating that this setting is processed somehow.

**Comment by timmcmic** [comment]
You are correct!   Also looks like a test case error.  The case scripts have both accept and reject - so it created a false positive in that the rights were maintained solely becuase something else had reject rights in the groups attribute.  I've fixed the source and am rerunning some isolated tests now.

**Comment by timmcmic** [comment]
FIxed 2.9.6.3 published today.
